spotbugs/src/main/java/edu/umd/cs/findbugs/TextUICommandLine.java

```diff
@@ -266,30 +270,40 @@ public boolean justPrintVersion() {
     private List<TextUIBugReporter> reporters = new ArrayList<>();
 
     /**
-     * Parse {@code optionExtraPart} and configure {@Link TextUIBugReporter} if it contains the
+     * Parse {@code optionExtraPart} and create a path-associated {@Link TextUIBugReporter} if it contains the
      * output file path such as {@code ":withMessages=path/to/file.extension"} and {@code "=/absolute/path/to/file.extension"}.
+     * Finally configure the created BugReporter with the {@code optionExtraPart}'s configuration information.
      *
-     * @param reporter the reporter to set a {@link PrintStream} based on the given file path
      * @param optionExtraPart extra part of the specified commandline option
-     * @return Remaining part of {@code optionExtraPart}
+     * @param ctor            A supplier for an unconfigured {@Link TextUIBugReporter}.
+     * @param handleOptions   A function that can configure the created {@code BugReporter} instance.
+     * @param <T>             The implementation type of the {@Link TextUIBugReporter} to propagate type
+     *                        information between {@link ctor} and {@link handleOptions}.
+     * @return The fully configured reporter, or {@code null}, if the reporter would output to a file that is already used as a reporter output file.
      */
-    /* visible for testing */ String handleOutputFilePath(TextUIBugReporter reporter, String optionExtraPart) {
+    /* visible for testing */ <T extends TextUIBugReporter> TextUIBugReporter initializeReporter(String optionExtraPart,
+            Supplier<T> ctor, BiConsumer<T, String> handleOptions) {
```

Comment:
It's a bit strange for me to have `optionExtraPart` as the first parameter, where it was the last one earlier and at the called functions it's the last one. However, since it has different type, it can't cause parameter mixup issues.